packages/@dcl/inspector/src/lib/data-layer/host/utils/engine-to-composite.ts

```diff
@@ -128,3 +130,85 @@ export function dumpEngineToCrdtCommands(engine: IEngine): Uint8Array {
 
   return crdtBuffer.toBinary()
 }
+
+/**
+ * Generate a TypeScript declaration file with a string literal union type containing all entity names in the scene.
+ * This allows for type-safe references to entities by name in scene scripts.
+ *
+ * @param engine The ECS engine instance
+ * @param outputPath The path where the .d.ts file should be written
+ * @param typeName The name for the generated type (default: "SceneEntityNames")
+ * @param fs FileSystem interface for writing the file
+ * @returns Promise that resolves when the file has been written
+ */
+export async function generateEntityNamesType(
+  engine: IEngine,
+  outputPath: string = 'scene-entity-names.d.ts',
+  typeName: string = 'SceneEntityNames',
+  fs: FileSystemInterface
+): Promise<void> {
+  try {
+    // Find the Name component definition
+
+    const NameComponent: typeof Name = engine.getComponentOrNull(Name.componentId) as typeof Name
+
+    if (!NameComponent) {
+      throw new Error('Name component not found in engine')
+    }
+
+    // Collect all names from entities
+    const names: string[] = []
+    for (const [_, nameValue] of engine.getEntitiesWith(NameComponent)) {
+      if (nameValue.value) {
+        names.push(nameValue.value)
+      }
+    }
+
+    // Sort names for consistency
+    names.sort()
+
+    // Remove duplicates
+    const uniqueNames = Array.from(new Set(names))
+
+    // Generate valid TypeScript identifiers
+    const validNames = uniqueNames.map((name) => {
+      // Create a valid TypeScript identifier
+      const validName = name
+        .replace(/[^a-zA-Z0-9_]/g, '_') // Replace non-alphanumeric chars with underscore
+        .replace(/^[0-9]/, '_$&') // Prepend underscore if starts with number
+
+      // Ensure no duplicate keys after validation
+      return { original: name, valid: validName }
+    })
```

Comment:
if the user creates two entities with names like: 'test-1' and 'test_1', the latter will be used as valid for both cases below on line 203. I think this may produce a bug?

tested it here: https://tsplay.dev/w2yyxW

Reply:
good catch, solved